reader.go

```diff
@@ -50,6 +58,31 @@ func (f *rsfReader) ReadSizeField(r io.Reader) (int, error) {
 	return int(sz), nil
 }
 
+func (f *rsfReader) ReadInt64Field(r io.Reader) (int64, error) {
+	bs := make([]byte, binary.MaxVarintLen64)
+	i, err := io.ReadFull(r, bs)
+	if err != nil {
+		return 0, err
+	} else if i != binary.MaxVarintLen64 {
+		return 0, fmt.Errorf("unexpected read size %d; expected %d", i, sizeFieldLen)
```

Comment:
This comment didn't go through for some reason, but this should be updated too:

```suggestion
		return 0, fmt.Errorf("unexpected read size %d; expected %d", i, binary.MaxVarintLen64)
```